http4s/server-circe/src/main/scala/endpoints/http4s/server/circe/JsonEntitiesFromCodec.scala

```diff
@@ -0,0 +1,44 @@
+package endpoints.http4s.server.circe
+
+import cats.implicits._
+import endpoints.algebra.Documentation
+import endpoints.http4s.server.Endpoints
+import io.circe.Json
+import io.circe.parser._
+import org.http4s
+import org.http4s.circe._
+import org.http4s.{DecodeFailure, InvalidMessageBodyFailure}
+
+trait JsonEntitiesFromCodec[F[_]]
+    extends Endpoints[F]
+    with endpoints.algebra.circe.JsonEntitiesFromCodec {
+
+  def jsonRequest[A](docs: Documentation = None)(
+      implicit codec: JsonRequest[A]): RequestEntity[A] =
+    req => {
+      def transform(t: Either[DecodeFailure, Json]): Either[DecodeFailure, A] =
+        for {
+          str <- t
+          a <- codec
+            .decode(str.toString)
```

Comment:
Why do we need to transform the `Json` value into a `String` value?

Reply:
My idea here was  to reuse decoder `jsonOf` from org.http4s.circe._ since it returns errors when it tries to decode invalid JSON request but I couldn't use `jsonOf[F, String]` because it requires payload to be String and not JSON object.
Because of that I'm decoding it into Circe's Json datatype and converting it to String to that I can use our `codec` to decode it to wanted type.

I'll see what can I do here to remove this intermediate step since it looks weird to me too 🙂